### PR TITLE
Pull Request for Issue 2464: onExportClicked not defined in AMSpectrumAndPeriodicTableView

### PR DESCRIPTION
**Pull request by mButlerDM**
Fixed a few issues with connections in AMSpectrumAndPeriodicTableView

onExportClicked() was connected in parent when it needed to be connected in the child classes.
Renamed onExportClicked and onSaveButtonClicked to onExportButtonClicked so child classes are consistent with each other.

Moved some connections that were made in the parent to the child so they would properly use the child's version of the slot function.

#2464

### CODE REVIEW

**Comment by mButlerDM** [comment]
@helfrij Could you review when you get a moment? Thanks!

File: source/ui/beamline/AMXRFDetailedDetectorView.cpp [hunk 1]
```diff
@@ -167,6 +172,8 @@ void AMXRFDetailedDetectorView::buildEnergyRangeSpinBoxView()
 {
 	AMSpectrumAndPeriodicTableView::buildEnergyRangeSpinBoxView();
 
+	connect(exportButton_, SIGNAL(clicked()), this, SLOT(onExportButtonClicked()));
```
Comment (helfrij) [inline source/ui/beamline/AMXRFDetailedDetectorView.cpp]: Should this connect statement be in the constructor with the others? Or is it better to keep it here?

**Comment by helfrij** [comment]
Looks good @mButlerDM!

Time Estimate: 5 minutes.

File: source/ui/beamline/AMXRFDetailedDetectorView.cpp (hunk 1 above)
Comment (XuetaoLiu-CLS) [inline source/ui/beamline/AMXRFDetailedDetectorView.cpp]: @mButlerDM @helfrij I think it might be better to be where we created the button. In this case it is the function `AMSpectrumAndPeriodicTableView::buildEnergyRangeSpinBoxView();`.

@mButlerDM if `onExportButtonClicked()` must be implemented in the Detailed detector view, you can make it a (pure) virtual slot in `AMSpectrumAndPeriodicTableView`.

**Comment by mButlerDM** [comment]
@helfrij  I think it makes more sense to have it somewhere else. I wasn't too what the best way to do that was. I followed @XuetaoLiu-CLS suggestion and used a pure virtual function declaration to keep it in the parent.

File: source/ui/beamline/AMXRFDetailedDetectorView.h [hunk 2]
```diff
@@ -48,13 +48,6 @@ class AMXRFDetailedDetectorView : public AMXRFBaseDetectorView
 	/// Re-implementing but still going to use the base class buildDetectorView since this view is merely adding to it.
 	virtual void buildDetectorView();
 
-	/// Returns the energy range used for filtering.  If no range has been provided the range is null.
-	const AMRange &energyRange() const { return emissionLineValidator_->range(); }
```
Comment (dretrex) [inline source/ui/beamline/AMXRFDetailedDetectorView.h]: Why were these removed?  Were they moved somewhere else and then not removed initially?

File: source/ui/dataman/AMSpectrumAndPeriodicTableView.h [hunk 3]
```diff
@@ -156,6 +156,8 @@ protected slots:
 
 	/// The export button.
 	QPushButton *exportButton_;
+	///
```
Comment (dretrex) [inline source/ui/dataman/AMSpectrumAndPeriodicTableView.h]: Why is this function buried deep within the protected members when it is supposed to be a slot?

Comment (dretrex) [inline source/ui/dataman/AMSpectrumAndPeriodicTableView.h]: Also, you need to be careful using virtual functions in constructors (I'm not sure about using them in connect statements).  In general, the use of virtual functions in constructors should be avoided because the virtual function won't be properly resolved using polymorphism yet.

**Comment by dretrex** [comment]
Time Estimate: 3 minutes

File: source/ui/beamline/AMXRFDetailedDetectorView.h (hunk 2 above)
Comment (mButlerDM) [inline source/ui/beamline/AMXRFDetailedDetectorView.h]: Yes they were supposed to be removed in the initial refactor, as I went over all the functions again I noticed they were defined in both AMSpectrumAndPeriodicTableView and here.

File: source/ui/dataman/AMSpectrumAndPeriodicTableView.h (hunk 3 above)
Comment (mButlerDM) [inline source/ui/dataman/AMSpectrumAndPeriodicTableView.h]: @dretrex Probably because I saw `exportButton_` and got excited, oops!

The connection to the virtual functions occurs in `buildEnergyRangeSpinBoxView()` which is called outside the constructor and after initialization in `AMXRFDetailedDetectorView`, though it is called in the constructor of `AMScanViewUtilities`. It might be worth refactoring the `AMScanViewUtilities` constructor to be more in line with AMXRF?
